src/crypto/hash.c

```diff
@@ -224,127 +223,125 @@ int sha512_hash(unsigned char *digest, const unsigned char *msg, size_t msg_size
-int sha3_512_hash(unsigned char *digest, const unsigned char *msg, size_t msg_size,
-                  const unsigned char *salt, size_t salt_size) {
+int shake256Hash(unsigned char* digest, size_t digest_size, const unsigned char* msg,
+                 size_t msg_size, const unsigned char* salt, size_t salt_size) {
     Keccak_HashInstance inst;
 
-    if(Keccak_HashInitialize_SHA3_512(&inst) == KECCAK_FAIL) {
+    if (Keccak_HashInitialize_SHAKE128(&inst) == KECCAK_FAIL) {
         return 1;
     }
 
-    return keccak_hash(digest, NULL, &inst, msg, msg_size, salt, salt_size);
+    return keccakHash(digest, &digest_size, &inst, msg, msg_size, salt, salt_size);
 }
 
-int shake128_hash(unsigned char *digest, size_t digest_size,
-                  const unsigned char *msg, size_t msg_size,
-                  const unsigned char *salt, size_t salt_size) {
-    Keccak_HashInstance inst;
+int kang12Hash(unsigned char* digest, size_t digest_size, const unsigned char* msg, size_t msg_size,
+               const unsigned char* salt, size_t salt_size) {
+    KangarooTwelve_Instance inst;
 
-    if(Keccak_HashInitialize_SHAKE128(&inst) == KECCAK_FAIL) {
+    if (KangarooTwelve_Initialize(&inst, digest_size)) {
         return 1;
     }
 
-    return keccak_hash(digest, &digest_size, &inst, msg, msg_size, salt, salt_size);
-}
+    if (KangarooTwelve_Update(&inst, msg, msg_size)) {
+        OPENSSL_cleanse(&inst, sizeof(inst));
+        return 1;
+    }
 
-int shake256_hash(unsigned char *digest, size_t digest_size,
-                  const unsigned char *msg, size_t msg_size,
-                  const unsigned char *salt, size_t salt_size) {
-    Keccak_HashInstance inst;
+    if (salt != NULL && KangarooTwelve_Update(&inst, salt, salt_size)) {
+        OPENSSL_cleanse(&inst, sizeof(inst));
+        return 1;
+    }
 
-    if(Keccak_HashInitialize_SHAKE128(&inst) == KECCAK_FAIL) {
+    if (KangarooTwelve_Final(&inst, digest, NULL, 0)) {
+        OPENSSL_cleanse(&inst, sizeof(inst));
         return 1;
     }
 
-    return keccak_hash(digest, &digest_size, &inst, msg, msg_size, salt, salt_size);
-}
+    OPENSSL_cleanse(&inst, sizeof(inst));
 
-int kang12_hash(unsigned char *digest, size_t digest_size, const unsigned char *msg, size_t msg_size,
-                const unsigned char *salt, size_t salt_size) {
-    KangarooTwelve_Instance inst;
+    return 0;
+}
 
-    if(KangarooTwelve_Initialize(&inst, digest_size)) {
+int keccakHash(unsigned char* digest, const size_t* digest_size, Keccak_HashInstance* inst,
+               const unsigned char* msg, size_t msg_size, const unsigned char* salt,
+               size_t salt_size) {
+    if (Keccak_HashUpdate(inst, msg, msg_size * 8) == KECCAK_FAIL) {
+        OPENSSL_cleanse(inst, sizeof(*inst));
         return 1;
     }
 
-    if(KangarooTwelve_Update(&inst, msg, msg_size)) {
-        OPENSSL_cleanse(&inst, sizeof(inst));
+    if (salt != NULL && Keccak_HashUpdate(inst, salt, salt_size * 8) == KECCAK_FAIL) {
```

Comment:
Whyd these lines flip from formatting?

Reply:
I think it only looks like it flipped because I moved the `keccakHash()` function to the bottom and that was just whatever the difference algorithm determined was most efficient